### DISCUSSION TITLE
Error: [ClientResponseError 0: Something went wrong while processing your request.]

### DISCUSSION THREAD

**Discussion by HeineZo** [pocketbase/pocketbase#1520] [Q&A] [answered]
I just started to implement Pocketbase into a personal project but I'm facing with an issue.
When I'm trying to list all the collections I have in my database, I got this error.
``` Error: [ClientResponseError 0: Something went wrong while processing your request.]```

When I log the attributes of the error it gives me that
```Error: {"url":"","status":0,"data":{},"isAbort":false,"originalError":{},"name":"ClientResponseError 0"}```

Here's my code
```js
import PocketBase from 'pocketbase';

const pb = new PocketBase('http://127.0.0.1:8090');

export async function loadPosts() {

	pb.collections.getFullList(200).then((result) => {
		// success...
		console.log('Result:', result);
	  }).catch((error) => {
		// error...
		console.log('Error:', JSON.stringify(error));
		console.log('Error:', error);
	  });
}
```

Also, I'm using Pocketbase in react-native with expo, all the packages are in the latest version.

Thank you for your help 🙌

**Comment by ganigeorgiev** [comment, marked as answer]
How and where are you running the react native app - in a device emulator? Could you confirm that you have access to the localhost where PocketBase is running?

You can try to log the original error in your `catch` block using `console.log(error.originalError)`.

Depending when and how `loadPosts()` is triggered, the error could be also a result of [autocancellation](https://github.com/pocketbase/js-sdk#auto-cancellation) (_`isAbort` usually would be `true` in this case but I'm not sure if in react native `DomException` is available_). Or in other words, try adding `pb.autoCancellation(false);` before the `getFullList()` call. If that's resolve the issue, then most likely `loadPosts()` is getting triggered multiple times without awaiting previous requests to complete.

**Reply by HeineZo** [reply]
I'm running the react native app on my iPhone with the Expo App and I changed ```localhost``` by ```http://127.0.0.1:8090``` but nothing changed.
I also add a ```pb.autoCancellation(false)``` before the ```getFullList()``` but nothing changed.

However thanks to the ```console.log(error.originalError)```, I discovered that it was a ```[TypeError: Network request failed]```.
Although I searched on the web with this error, I didn't see something that could help me.

Any idea ?

**Reply by ganigeorgiev** [reply]
It seems that your PC localhost (where PocketBase is running) is not accessible from your device.

You have to use either your local network IP (if both devices are connected to the same network) OR use a tool like ngrok to create a public url for your localhost - https://ngrok.com/.

**Reply by HeineZo** [reply]
Thanks it worked! I uploaded my pocketbase database on [fly.io](https://fly.io/) and I have now access to it everywhere 🥳

**Reply by chiatzeheng** [reply]
Hello im facing the same issue with expo and react native as well how would i be able to use local network IP

**Reply by ganigeorgiev** [reply]
@tiamat9 You have to locate your local network IP (eg, `ifconfig`) and use that address as connection url to PocketBase, but this may not always work since it depends on your local firewall/network configuration and may require further setup. The easiest way to access your localhost from an external device is to use a ssh tunneling service like [ngrok](https://ngrok.com/).

**Reply by chiatzeheng** [reply]
Thanks now it works!

**Comment by wesleywcr** [comment]
Changing http://127.0.0.1:8090 to http://10.0.2.2:8090 worked here
